moveit_simple/src/moveit_simple.cpp

```diff
@@ -619,46 +671,157 @@ void OnlineRobot::execute(const std::string traj_name, bool collision_check)
   bool success = false;
   if ( traj_info_map_.count(traj_name) )
   {
-    control_msgs::FollowJointTrajectoryGoal goal;
+    JointTrajectoryType goal;
     goal.trajectory.joint_names = joint_group_->getVariableNames();
     try
     {
       if ( toJointTrajectory(traj_name, goal.trajectory.points, collision_check) )
       {
+
+        // Modify the speed of execution for the trajectory based off of the speed_modifier_
+        for (std::size_t i = 0; i < goal.trajectory.points.size(); i++)
+        {
+          goal.trajectory.points[i].time_from_start *= (1.0/speed_modifier_);
+        }
+
         ros::Duration traj_time =
             goal.trajectory.points[goal.trajectory.points.size()-1].time_from_start;
         ros::Duration timeout(TIMEOUT_SCALE * traj_time.toSec());
         if (action_.sendGoalAndWait(goal, timeout) == actionlib::SimpleClientGoalState::SUCCEEDED)
         {
-          ROS_INFO_STREAM("Successfully executed trajectory: " << traj_name);
+          ROS_INFO_STREAM("Successfully executed trajectory: [" << traj_name << "]");
         }
         else
         {
-          ROS_ERROR_STREAM("Trajectory " << traj_name << " failed to exectue");
+          ROS_ERROR_STREAM("Trajectory [" << traj_name << "] failed to exectue");
           throw ExecutionFailureException("Execution failed for "+ traj_name);
         }
       }
       else
       {
-        ROS_ERROR_STREAM("Failed to convert " << traj_name << " to joint trajectory");
+        ROS_ERROR_STREAM("Failed to convert [" << traj_name << "] to joint trajectory");
         throw IKFailException("Conversion to joint trajectory failed for " + traj_name);
       }
     }
     catch(CollisionDetected &cd)
     {
-      ROS_ERROR_STREAM("Collision detected in the " << traj_name);
+      ROS_ERROR_STREAM("Collision detected in the [" << traj_name << "]");
       throw cd;
     }
   }
   else
   {
-    ROS_ERROR_STREAM("Trajectoy " << traj_name << " not found");
+    ROS_ERROR_STREAM("Trajectoy [" << traj_name << "] not found");
     throw std::invalid_argument("No trajectory found named " + traj_name);
   }
 }
 
 
 
+void OnlineRobot::execute(JointTrajectoryType & goal,
```

Comment:
The method should simply check all points for collision, then populate the action goal and send.  If there is any collision, then the method should throw.  The structure below does not have to be that complex.